### PR TITLE
User checker refactoring aa

### PR DESCRIPTION
**Pull request by ArtemAnoshin**

### CODE REVIEW

File: lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php [hunk 1]
```diff
@@ -208,27 +201,29 @@ public static function ctAjaxCheckUsers()
...
 
         $wc_orders = '';
 
         if ( $wc_active && ! empty($_POST['accurate_check']) ) {
-            $wc_orders = " AND NOT EXISTS (SELECT posts.* FROM {$wpdb->posts} AS posts INNER JOIN {$wpdb->postmeta} AS postmeta WHERE posts.post_type = 'shop_order' AND posts.post_status = 'wc-completed' AND posts.ID = postmeta.post_id AND postmeta.meta_key = '_customer_user' AND postmeta.meta_value = {$wpdb->users}.ID)";
+        	$sql_command = 'WHERE ';
+        	if ($from_till) {
+		        $sql_command = ' AND ';
+	        }
+            $wc_orders = $sql_command . "NOT EXISTS (SELECT posts.* FROM {$wpdb->posts} AS posts INNER JOIN {$wpdb->postmeta} AS postmeta WHERE posts.post_type = 'shop_order' AND posts.post_status = 'wc-completed' AND posts.ID = postmeta.post_id AND postmeta.meta_key = '_customer_user' AND postmeta.meta_value = {$wpdb->users}.ID)";
```
Comment (safronik) [inline lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php:214]: Please, split this in a few rows using concatenation operator.

Comment (ArtemAnoshin) [reply lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php]: done

**Review by Glomberg — CHANGES_REQUESTED** [review]

File: js/src/cleantalk-users-checkspam.js [hunk 2]
```diff
@@ -195,6 +199,10 @@ function ct_send_users(){
 						status_string += ctUsersCheck.ct_status_string_warning;
 					jQuery('#ct_checking_status').html(status_string);
 					jQuery('#ct_error_message').hide();
+
+					var offset = Number(getCookie('apbct_check_users_offset')) + 100;
+					document.cookie = 'apbct_check_users_offset' + "=" + offset + "; path=/";
```
[the remark this answers is not in the document]
Comment (ArtemAnoshin) [reply js/src/cleantalk-users-checkspam.js:204]: done

File: lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php [hunk 3]
```diff
@@ -208,27 +186,26 @@ public static function ctAjaxCheckUsers()
...
 
         $wc_orders = '';
 
         if ( $wc_active && ! empty($_POST['accurate_check']) ) {
-            $wc_orders = " AND NOT EXISTS (SELECT posts.* FROM {$wpdb->posts} AS posts INNER JOIN {$wpdb->postmeta} AS postmeta WHERE posts.post_type = 'shop_order' AND posts.post_status = 'wc-completed' AND posts.ID = postmeta.post_id AND postmeta.meta_key = '_customer_user' AND postmeta.meta_value = {$wpdb->users}.ID)";
+            $sql_command = $from_till ? ' AND ' : 'WHERE ';
+            $wc_orders = $sql_command . "NOT EXISTS (SELECT posts.* FROM {$wpdb->posts} AS posts INNER JOIN {$wpdb->postmeta} AS postmeta WHERE posts.post_type = 'shop_order' AND posts.post_status = 'wc-completed' AND posts.ID = postmeta.post_id AND postmeta.meta_key = '_customer_user' AND postmeta.meta_value = {$wpdb->users}.ID)";
         }
 
+        $offset = $_COOKIE['apbct_check_users_offset'] ?: 0; // TODO: implement alternative cookies
```
[the remark this answers is not in the document]
Comment (ArtemAnoshin) [reply lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php:199]: ok

File: lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php [hunk 4]
```diff
@@ -208,27 +186,26 @@ public static function ctAjaxCheckUsers()
...
-			WHERE
-				NOT EXISTS(SELECT * FROM {$wpdb->usermeta} as meta WHERE {$wpdb->users}.ID = meta.user_id AND meta.meta_key = 'ct_bad') AND
-		        NOT EXISTS(SELECT * FROM {$wpdb->usermeta} as meta WHERE {$wpdb->users}.ID = meta.user_id AND meta.meta_key = 'ct_checked') AND
-		        NOT EXISTS(SELECT * FROM {$wpdb->usermeta} as meta WHERE {$wpdb->users}.ID = meta.user_id AND meta.meta_key = 'ct_checked_now')
 		        $wc_orders
 			    $from_till
-			ORDER BY {$wpdb->users}.user_registered ASC
-			LIMIT $amount;"
+			ORDER BY {$wpdb->users}.ID ASC
+			LIMIT $amount OFFSET $offset;"
```
[the remark this answers is not in the document]
Comment (ArtemAnoshin) [reply lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php:208]: ok

**Review by Glomberg — APPROVED** [review]

File: lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php (hunk 1 above)
Comment (safronik) [reply lib/Cleantalk/ApbctWP/FindSpam/UsersChecker.php:199]: I don't see any changes in the sql-string request. 

Please, split the long sql-request string in a few rows using concatenation operator. Similar to this one https://github.com/CleanTalk/security-malware-firewall/blob/master/inc/spbc-scanner.php#L487. It's hard to read.

**Pull request merged**